tox.ini

```diff
@@ -12,16 +10,18 @@ basepython =
 
 deps =
     -r{toxinidir}/requirements/tox.txt
-    {py27}-django111: Django>=1.11,<2.0
-    {py39}-django22: Django>=2.2,<3.0
-    {py39}-django31: Django>=3.1,<3.2
-    {py39}-django32: Django>=3.2,<3.3
```

Comment:
what is the difference between the `{....}` variants and the ones without? and why are `{py39}` and `{py27}` in the hunk above not transformed in the same manner?

Reply:
tox allows to only do certain things for certain environments. By putting something between curly braces in `envlist` you can use the thing between curly braces to do it for all environments where that element was chosen. in the `basepython` section we use this to choose the python version:
```ini
basepython =
    py27: python2.7
    py39: python3.9
```

If we'd remove the curly braces we could not do that anymore and we'd have to specify the python version for all environments separatly.